lighthouse-core/test/lib/network-request-test.js

```diff
@@ -0,0 +1,243 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const NetworkRequest = require('../../lib/network-request');
+const assert = require('assert');
+
+/* eslint-env jest */
+describe('network request', function() {
+  function getRequest() {
+    const req = new NetworkRequest();
+    req.transferSize = 100;
+    req.responseHeaders = [{name: NetworkRequest.HEADER_FETCHED_SIZE, value: 10}];
+    return req;
+  }
+
+  describe('update transfer size for lightrider', function() {
+    it('does nothing if not Lightrider', function() {
+      const req = getRequest();
+      global.isLightrider = false;
+
+      assert.equal(req.transferSize, 100);
+      req._updateTransferSizeForLightrider();
```

Comment:
ideally we shouldn't be testing the internal implementation.

How do you feel about using `network-records-to-devtools-log.js`? It already supports headers, so should be pretty easy to change this code over to using that. It would create the devtools log from a minimal example network record with `X-*` response headers, run the log through `NetworkRecorder.recordsFromLogs`, and then assert on the record that comes out.

Reply:
the change really should be pretty trivial. I can add it if you want, so we can get this in faster

Reply:
Yeah can you get that.  I'm not 100% on what it entails.